tests/test_misc.py

```diff
@@ -0,0 +1,49 @@
+import pandas as pd
+from pandas._testing import assert_frame_equal
+
+from core.validation.utils import find_null_values, remove_duplicate_indexes
+
+
+def test_remove_duplicate_indexes():
+    df = pd.DataFrame(
+        index=[1, 1, 2, 2, 3],
+        data=[
+            {"Project ID": "AB-ABC-01"},
+            {"Project ID": "AB-ABC-02"},  # should be removed
+            {"Project ID": "AB-ABC-03"},
+            {"Project ID": "AB-ABC-04"},  # should be removed
+            {"Project ID": "AB-ABC-05"},
+        ],
+    )
+    df = remove_duplicate_indexes(df)
+    expected_df = pd.DataFrame(
+        index=[1, 2, 3],
+        data=[
+            {"Project ID": "AB-ABC-01"},
+            {"Project ID": "AB-ABC-03"},
+            {"Project ID": "AB-ABC-05"},
+        ],
+    )
+    assert_frame_equal(df, expected_df)
+
+
+def test_find_null_values():
+    df = pd.DataFrame(
+        index=[7, 9],
+        data=[
+            {"not-nullable": "", "nullable": ""},
+            {"not-nullable": "random input", "nullable": ""},
+        ],
+    )
+    column = "not-nullable"
+
+    invalid_rows = find_null_values(df, column)
+
+    expected_invalid_rows = pd.DataFrame(
+        index=[7],
+        data=[
+            {"not-nullable": "", "nullable": ""},
+        ],
+    )
+
+    assert_frame_equal(invalid_rows, expected_invalid_rows)
```

Comment:
These are best placed in `tests/validation_tests/test_utils` instead I think

Reply:
Makes sense, I've moved the tests and pushed